qiskit-superstaq/qiskit_superstaq/superstaq_job.py

```diff
@@ -90,35 +98,85 @@ def result(self, timeout: Optional[float] = None, wait: float = 5) -> qiskit.res
             }
         )
 
-    def status(self) -> qiskit.providers.jobstatus.JobStatus:
-        """Query for the job status."""
+    def _check_if_stopped(self) -> None:
+        """Verifies that the job status is not in a canceled state and raises an
+        exception if it is.
 
-        job_id_list = self._job_id.split(",")  # separate aggregated job ids
+        Raises:
+            SuperstaQUnsuccessfulJob: If the job been canceled or an error has occured.
+            SuperstaQException: If unable to get the status of the job from the API.
+        """
+        if self._overall_status == "Canceled":
+            raise gss.superstaq_exceptions.SuperstaQUnsuccessfulJobException(
+                self._job_id, self._overall_status
+            )
+
+    def _refresh_job(self) -> None:
+        """Queries the server for the current job status"""
 
-        status = "Done"
+        # When we have multiple jobs, we will take the "worst status" among the jobs
+        # For example, if object of the jobs are still queued/canceled, we report Queued/Cancelled
+        # as the status for the entire batch.
+        # The overall status will only be reported as Done if all jobs are done
+        # following the "worst status" rule.
+
+        job_id_list = self._job_id.split(",")  # separate aggregated job ids
 
-        # when we have multiple jobs, we will take the "worst status" among the jobs
-        # For example, if any of the jobs are still queued, we report Queued as the status
-        # for the entire batch.
         for job_id in job_id_list:
-            result = self._backend._provider._client.get_job(job_id)
-            temp_status = result["status"]
 
+            if job_id in self._job_info and self._job_info[job_id] not in self.TERMINAL_STATES:
+                result = self._backend._provider._client.get_job(job_id)
+                self._job_info[job_id] = result
+
+            if job_id not in self._job_info:
+                result = self._backend._provider._client.get_job(job_id)
+                self._job_info[job_id] = result
+
+            temp_status = self._job_info[job_id]["status"]
             if temp_status == "Queued":
-                status = "Queued"
+                self._overall_status = "Queued"
                 break
+            elif temp_status == "Canceled":
+                self._overall_status = "Canceled"
+                break
+            elif temp_status == "Error":
+                self._overall_status = "Error"
+                raise qiskit.providers.JobError(
```

Comment:
Yeah, qiskit has its job error, but I kept that error because it was present in the previous implementation if the `status == "Error"` when calling `_wait_for_results`. I thought that maybe the specific messaging of ` "API returned error:\n" + str(self._job_info[job_id]` was wanted when using Superstaq. Removed now to reduce redundancy.